tests/electron/electron-multi-window-app.spec.ts

```diff
@@ -0,0 +1,41 @@
+/**
+ * Copyright (c) Microsoft Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the 'License');
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an 'AS IS' BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import path from 'path';
+import { electronTest as test, expect } from './electronTest';
+
+test('should support multi-window electron app with requestSingleInstanceLock', async ({
+  playwright,
+}) => {
+  const firstApp = await playwright._electron.launch({
+    args: [path.join(__dirname, 'electron-multi-window-app.js')],
+  });
+  expect(firstApp.windows()).toHaveLength(1);
+  const firstWindow = firstApp.windows()[0];
+  await expect(firstWindow).toHaveTitle('Window 1');
+  try {
```

Comment:
This test will be flaky / failing because `window` will be sometimes dispatched before `launch` completes. You should do something like this instead:

```js
  await Promise.all([
    firstApp.waitForEvent('window'),
    playwright._electron.launch({
      args: [path.join(__dirname, 'electron-multi-window-app.js')],
    })
  ]);
```

which reads is 'expect launching second instance of app to create a window in the first instance'.

Reply:
Hm, Promise.all rejects as soon as  `playwright._electron.launch` rejects. 

![test-error](https://user-images.githubusercontent.com/23744935/159761314-bdf5a072-86e4-4d8d-838c-52e0fecb029e.png)

But I like the idea. What do you think about this version instead?

```js
const secondWindowPromise = firstApp.waitForEvent('window')
try {
  await playwright._electron.launch({
      args: [path.join(__dirname, 'electron-multi-window-app.js')],
    })
} catch (error) {
  if (error.message !== 'electron.launch: canceled') throw error;
}
await secondWindowPromise
expect(firstApp.windows()).toHaveLength(2);
```
![other-version](https://user-images.githubusercontent.com/23744935/159761949-4695033e-85c2-4702-92c2-c62dcf93b4a0.png)